### DISCUSSION TITLE
Shredding data for individual dates / sub-events of an event series

### DISCUSSION THREAD

**Discussion by literarymachine** [pretix/pretix#2421] [Q&A] [answered]
Hi, and thanks for pretix! It really is a great tool, and although it can be quite complex, most things I can figure out myself. When it comes to shredding data though, I am a bit lost. I have a long running series of events and have the requirement to delete personal data at some point. Is there any way to shred data for individual dates in the series? After all, these are basically sub-events, but I can't find the data shredder for sub-events, only for the top level event which will potentially run indefinitely.

**Comment by raphaelm** [comment, marked as answer]
Hi there! We understand this is a problem and unfortunately do not have a great solution for it right now. The "shredder" framework included in pretix core doesn't really allow to easily plug in this workload.

We have a plugin pretix-coronashredder used by some of our Hosted and Enterprise customers which does something like this, but isn't really configurable and only suitable for specific use cases. We're currently not releasing this plugin publicly for reasons I can't really go into at this time, but we're giving it out at no additional charge to everyone having some kind of contractual relationship (support contract, license for other plugins, …) with us.

Basically what it does is scrubbing all data from the pretixbase_orderposition, pretixbase_questionanswer, and pretixbase_logentry table if there's a reference to a subevent that is more than X days in the past.
Right now the only solutions for open source pretix would be doing this e.g. with an SQL level (or cycling through event series every few months).

**Reply by literarymachine** [reply]
Thanks @raphaelm, we'll look into those options!
